config/testreport/js/xlt.js

```diff
@@ -564,6 +564,231 @@
             });
         })();
 
+        // creates the echarts but defers their full setup
+        (function setupECharts() {
+            $('div.charts div.echart').each(function () {
+                // Loads the chart data and completes the echart setup.
+                function setUpEChart(echart, name, url) {
+                    $.getJSON({
+                        url: url,
+                        data: null,
+                        beforeSend: function (xhr) {
+                            // avoid errors in browser console (XML Parsing Error: syntax error) when loading from file system
+                            xhr.overrideMimeType("application/json");
+                        },
+                        success: function (data) {
+                            echart.hideLoading();
+
+                            // create time series data
+                            var dataMean = data.map(item => [item[0], item[1]]);                    // timestamp and mean value
+                            var dataMinimum = data.map(item => [item[0], item[2]]);                 // timestamp and min value
+                            var dataMaximum = data.map(item => [item[0], item[3]]);                 // timestamp and max value
+                            var dataMaxMinDiff = data.map(item => [item[0], item[3] - item[2]]);    // timestamp and diff value
```

Comment:
Instead of iterating over the (potentially large amount of) data multiple times, we may optimize this a little bit and iterate over it only once using a simple for-of loop.

```suggestion
                            var dataMean = [];
                            var dataMinimum = [];
                            var dataMaximum = [];
                            var dataMinMaxDiff = [];

                            // create time series data
                            for (var item of data) {
                                // timestamp and mean value
                                dataMean.push([item[0], item[1]]);
                                // timestamp and min value
                                dataMinimum.push([item[0], item[2]]);
                                // timestamp and max value
                                dataMaximum.push([item[0], item[3]]);
                                // timestamp and diff value
                                dataMaxMinDiff.push([item[0], item[3] - item[2]]);
                            }
```

Reply:
Done.